File: android/src/main/kotlin/com/pdfbarcode/pdf_barcode_decoder/PdfBarcodeDecoderPlugin.kt
package com.pdfbarcode.pdf_barcode_decoder

import android.content.Context
import android.os.Handler
import android.os.Looper
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import java.util.concurrent.Executors

class PdfBarcodeDecoderPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel
    private lateinit var context: Context
    private val executor = Executors.newCachedThreadPool()
    private val mainHandler = Handler(Looper.getMainLooper())

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "pdf_barcode_decoder")
        channel.setMethodCallHandler(this)
        context = flutterPluginBinding.applicationContext
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        if (call.method == "decodePdf") {
            val pdfBytes = call.argument<ByteArray>("pdfBytes")
            val filePath = call.argument<String>("filePath")
            val config = call.argument<Map<String, Any>>("config") ?: emptyMap()

            executor.execute {
                try {
                    val manager = PdfDecodeManager(context)
                    val barcodes = manager.decodePdf(pdfBytes, filePath, config)
                    mainHandler.post {
                        result.success(barcodes)
                    }
                } catch (e: Exception) {
                    val message = e.message ?: "Unknown error"
                    val code = if (message.contains(":")) message.substringBefore(":") else "RENDER_FAILED"
                    val details = if (message.contains(":")) message.substringAfter(":").trim() else message
                    mainHandler.post {
                        result.error(code, details, null)
                    }
                }
            }
        } else {
            result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}

File: ios/Classes/PdfBarcodeDecoderPlugin.swift
import Flutter
import UIKit

public class PdfBarcodeDecoderPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "pdf_barcode_decoder", binaryMessenger: registrar.messenger())
    let instance = PdfBarcodeDecoderPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    if call.method == "decodePdf" {
        guard let args = call.arguments as? [String: Any] else {
            result(FlutterError(code: "INVALID_ARGS", message: "Arguments must be a dictionary", details: nil))
            return
        }
        
        let pdfBytes = args["pdfBytes"] as? FlutterStandardTypedData
        let filePath = args["filePath"] as? String
        let config = args["config"] as? [String: Any] ?? [:]
        
        DispatchQueue.global(qos: .userInitiated).async {
            let manager = PdfDecodeManager()
            do {
                let barcodes = try manager.decodePdf(pdfBytes: pdfBytes, filePath: filePath, config: config)
                DispatchQueue.main.async {
                    result(barcodes)
                }
            } catch let error as NSError {
                let code = error.domain == "INVALID_PDF" || error.domain == "ENCRYPTED_PDF" ? error.domain : "RENDER_FAILED"
                DispatchQueue.main.async {
                    result(FlutterError(code: code, message: error.localizedDescription, details: nil))
                }
            }
        }
    } else {
        result(FlutterMethodNotImplemented)
    }
  }
}
